src/rhsm/utils.py

```diff
@@ -335,8 +320,16 @@ def __init__(self, description: Optional[str]):
         config = rhsm.config.get_config_parser()
         if config.get("rhsm", "progress_messages") == "0":
             self.quiet = True
-        if not inj.require(inj.PROGRESS_MESSAGES):
-            self.quiet = True
+        with contextlib.suppress(AttributeError, KeyError):
```

Comment:
hmm instead of suppressing errors, wouldn't it be better to only check whether `inj is not None`?

Reply:
Hmm the situation "`injection` with no `PROGRESS_MESSAGES`" is basically what was the situation before the addition of the progress messages (and `PROGRESS_MESSAGES`); then the progress messages (and `PROGRESS_MESSAGES`) were added, and that's the `import` issue being fixed there; hence, after that, `injection` definitely will have `PROGRESS_MESSAGES` feature.

So, unless I missed anything, if `injection` is not None, then it will have `PROGRESS_MESSAGES` too, as this source and what provides `injection` are part of the same sources (so they are both updated at the same time, there is no "disalignment").

Reply:
Oh, I missed something: the code in `rhsm` does not deal with `injection` at all (rightfully, it is in a different module), so the `require()` raises `KeyError` because `injection` is not initialized :-/ Drat.

So we need to catch `KeyError` because `injection` can not have the element for `PROGRESS_MESSAGES`. I do think it's still better to check for `inj` rather than suppressing `AttributeError`: it's a known situation (imported `injection` or not), and we don't want to blindly suppress a class of issues (which might hide potential problems in the future).

Reply:
Why do you use `contextlib.suppress`? Why don't you use following approach:

```python
if inj is not None:
    if hasattr(inj, `PROGRESS_MESSAGES`):
        if not inj.require(inj.PROGRESS_MESSAGES):
            self.quiet = True   
```